tests/test_writer.py

```diff
@@ -2484,6 +2484,16 @@ def test_append_pdf_with_dest_without_page(caplog):
     assert len(writer.named_destinations) == 3
 
 
+@pytest.mark.enable_socket
+def test_destination_is_none():
+    """Tests for PR#2963"""
+    url = "https://raw.githubusercontent.com/dxsooo/pypdf/refs/heads/main/resources/3.pdf"
```

Comment:
Please upload the file to a comment here instead and use the link from it. Otherwise, we rely on your fork to be available.

Reply:
Additionally, it seems like you mostly copied the code from my latest PR. Does this really match your issue as well?

Reply:
ok, wait a minute

Reply:
> Additionally, it seems like you mostly copied the code from my latest PR. Does this really match your issue as well?

not the same, my case happens in some page of a valid file, nothing about the file annotations

Reply:
If there is an easier way to reproduce it (for example by loading the file into a PdfReader directly and accessing some attributes), I would recommend doing this. What was the original code that triggered your issue?

Reply:
> If there is an easier way to reproduce it (for example by loading the file into a PdfReader directly and accessing some attributes), I would recommend doing this. What was the original code that triggered your issue?

I just found this when I was merging files ...